crates/rspack_loader_preact_refresh/src/lib.rs

```diff
@@ -27,13 +27,14 @@ impl PreactRefreshLoader {
 #[async_trait::async_trait]
 impl Loader<RunnerContext> for PreactRefreshLoader {
   async fn run(&self, loader_context: &mut LoaderContext<RunnerContext>) -> Result<()> {
-    let content = loader_context
-      .take_content()
-      .expect("Content should be available");
+    let Some(content) = loader_context.take_content() else {
+      return Ok(());
+    };
     let mut source = content.try_into_string()?;
     source += "\n";
     source += include_str!("runtime.js");
```

Comment:
Enhanced string concatenation for improved performance. For detailed benchmarks and alternative methods, refer to: [Hoodie's String Concatenation Benchmarks](https://github.com/hoodie/concatenation_benchmarks-rs)

Reply:
It's out of the scope of current PR. But you can create an issue though.